pmpro-sponsored-members.php

```diff
@@ -327,9 +328,43 @@ function pmprosm_sponsored_account_change( $level_id, $user_id ) {
 	else
 		$seats = "";
 
+	$level = pmpro_getLevel( $level_id );
+
 	$sqlQuery = "UPDATE $wpdb->pmpro_discount_codes SET uses = '" . $seats . "' WHERE id = '" . $code_id . "' LIMIT 1";
 	$wpdb->query( $sqlQuery );
 
+	//Check if its a renewal - if so, extend its validity
+	$morder = new MemberOrder();
+	$morder->getLastMemberOrder( $user_id, 'success', $level_id );
+	if( $morder->is_renewal() ) {
+		//Is a renewal, so we should extend the discount code and reset the uses.
+		if( ! empty( $level->cycle_period ) ){
+			//Is a recurring level
+			$discount_code = pmprosm_getDiscountCodeByCodeID( $code_id );
+			
+			if( ! empty( $discount_code ) ) {
+				$expiration = $discount_code->expires;
+				//We're extending the uses instead of resetting this
+				//Will need to test this a bit more for real-world use cases
+				$uses = $discount_code->uses;
+
+				$new_expiration_date = date('Y-m-d', strtotime( $expires . ' + '.$level->cycle_number.' '.$level->cycle_period ) );
+
+				$updated = $wpdb->update( $wpdb->pmpro_discount_codes, 
+					array (
+						'uses' => esc_sql( $uses + $seats ),
```

Comment:
Do sites want to increase the amount of child seats that a parent has when the parent renews? I would imagine that many sites would want to keep the number of seats the same, similar to if they were using automatically recurring payments and the parent was just renewing their membership.